### ISSUE TITLE
problems with unquoting subtrees into trees

### ISSUE THREAD

**Issue by xeno-by** [scalameta/scalameta#223]
We have several places in the current implementation of quasiquotes, where unquoting a subtree into a tree is not as easy as writing `... $subtree ...`. We need to scan through the spec, detect all those cases and think what to do with them.

Examples:
- `new $template` doesn't work
- `class C $ctor` or something doesn't work

**Comment by xeno-by** [comment]
Also see #405.

**Comment by dwijnand** [comment]
Ah, good at least to get confirmation that it's truly impossible to define a Term.New. In my code it's one of the few parts that doesn't use quasiquotes but:

```scala
    val ctorref = Ctor.Ref.Name(tname.value)
    val ctorNew = Term.New(Template(Nil, sciSeq(q"$ctorref(..$aexprs)"), anonParam, None))
``

**Comment by xeno-by** [comment]
@dwijnand What exactly would you like to achieve with your code snippet? I have a strong suspicion that your use case will be supported in 2.0.

**Comment by dwijnand** [comment]
Defining a variant of `case class` via macro annotation: https://github.com/dwijnand/dataclasses/blob/e244272b58a5d5abd90a1b6fcd3ca561d17d9766/src/main/scala/dataclasses/internal/DataMacros.scala#L40-L44

**Comment by olafurpg** [comment]
The next release introduces Term.Init, which avoids the need for templates in most basic cases.

```scala
scala> init"Foo"
res3: scala.meta.Init = Foo

scala> q"new $res3"
res4: meta.Term.New = new Foo

// templates are necessary in Term.NewAnonymous
scala> q"new $res5 { val x = 2 }"
res9: meta.Term.NewAnonymous = new Foo(1, 2) { val x = 2 }
```

**Comment by xeno-by** [comment]
@olafurpg and I had a discussion, and we decided to pronounce quasiquotes feature-complete.

There are still some open issues (#46, #223, #251, #280 and others), but they all have two things in common: 1) they require non-trivial design work to be fixed, 2) they aren't popular at all. Given our recent decision to start avoiding unwarranted breaking changes in 2.0, we'll consider quasiquotes good enough at this point and will close all outstanding feature requests.

We will still be fixing bugs (in fact, I've fixed about a dozen of such bugs for 2.0), but without strong evidence our design efforts will be directed elsewhere.
